src/API/Reports/Cache.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * REST API Reports Cache.
+ *
+ * Handles report data object caching.
+ *
+ * @package WooCommerce Admin/API
+ */
+
+namespace Automattic\WooCommerce\Admin\API\Reports;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Reports Cache class.
+ *
+ * @package WooCommerce Admin/API
+ */
+class Cache {
+	/**
+	 * Cache version. Used to invalidate all cached values.
+	 */
+	const VERSION_OPTION = 'woocommerce_reports_cache_version';
+
+	/**
+	 * Default version number rollover threshold.
+	 */
+	const DEFAULT_VERSION_LIMIT = 99999;
+
+	/**
+	 * Increase cache version number.
+	 *
+	 * @return bool
+	 */
+	public static function bump_version() {
+		/**
+		 * Upper limit to roll cache version back to zero.
+		 *
+		 * @param int $limit Limit.
+		 */
+		$limit   = apply_filters( 'woocommerce_reports_cache_version_limit', self::DEFAULT_VERSION_LIMIT );
+		$limit   = is_numeric( $limit ) ? (int) $limit : self::DEFAULT_VERSION_LIMIT;
+		$version = ( self::get_version() + 1 ) % $limit;
+
+		return update_option( self::VERSION_OPTION, $version );
```

Comment:
This could probably be simplified by using a hash based on timestamp? If the version mismatches the stored data, it's invalidated. It doesn't necessarily need to be a number that increments.